frontend/src/features/mapSideBar/MotiInventoryContainer.tsx

```diff
@@ -34,163 +29,180 @@ export interface IMotiInventoryContainerProps {
-    key: InventoryTabNames.value,
-    name: 'Value',
-  });
-
-  var defaultTab = InventoryTabNames.title;
-
-  if (showPropertyInfoTab) {
-    if (props.readOnly) {
-      tabViews.push({
-        content: <PropertyDetailsTabView property={propertyViewForm} loading={propertyLoading} />,
-        key: InventoryTabNames.property,
-        name: 'Property Details',
-      });
+  const handleSaveClick = async () => {
+    if (formikRef !== undefined) {
+      formikRef.current?.setSubmitting(true);
+      formikRef.current?.submitForm();
+    }
+  };
+
+  const handleCancelClick = () => {
+    if (formikRef !== undefined) {
+      if (formikRef.current?.dirty) {
+        setShowCancelConfirmModal(true);
+      } else {
+        handleCancelConfirm();
+      }
     } else {
-      tabViews.push({
-        content: <UpdatePropertyDetailsContainer pid={props.pid} />,
-        key: InventoryTabNames.property,
-        name: 'Property Details',
-      });
+      handleCancelConfirm();
     }
-    defaultTab = InventoryTabNames.property;
-  }
-
-  if (propertyAssociations?.id !== undefined) {
-    tabViews.push({
-      content: (
-        <PropertyAssociationTabView
-          isLoading={propertyAssociationsLoading}
-          associations={propertyAssociations}
-        />
-      ),
-      key: InventoryTabNames.pims,
-      name: 'PIMS Files',
-    });
-  }
-
-  const onEditPropertyInformation = useCallback(
-    () => history.push(`/mapview/property/${props.pid}/edit`),
-    [history, props.pid],
-  );
+  };
+
+  const handleCancelConfirm = () => {
+    if (formikRef !== undefined) {
+      formikRef.current?.resetForm();
+    }
+    setIsEditing(false);
+    setShowCancelConfirmModal(false);
+  };
 
   return (
     <MapSideBarLayout
       title="Property Information"
-      header={
-        <MotiInventoryHeader
-          ltsaData={ltsaData}
-          ltsaLoading={ltsaLoading}
-          propertyLoading={propertyLoading}
-          property={apiProperty}
-          showEditButton={props.readOnly}
-          onEdit={onEditPropertyInformation}
-          onZoom={props.onZoom}
-        />
+      header={<MotiInventoryHeader composedProperty={composedProperty} onZoom={props.onZoom} />}
+      footer={
+        isEditing && (
+          <SidebarFooter
+            isOkDisabled={formikRef?.current?.isSubmitting}
+            onSave={handleSaveClick}
+            onCancel={handleCancelClick}
+          />
+        )
       }
       icon={<LotIcon className="mr-1" />}
       showCloseButton
       onClose={props.onClose}
     >
-      <InventoryTabs
-        tabViews={tabViews}
-        defaultTabKey={defaultTab}
-        activeTab={activeTab}
-        setActiveTab={setActiveTab}
-      />
+      <>
+        <PropertyViewSelector
+          composedProperty={composedProperty}
+          isEditMode={isEditing}
+          setEditMode={setIsEditing}
+          setFormikRef={setFormikRef}
+          onSuccess={onSuccess}
+        />
+        <GenericModal
```

Comment:
I wonder if we should register some of these common ones somewhere at the app level, and then create a context to control their display. That way we wouldn't have to keep registering the modals in various containers/forms throughout the app.

Reply:
Thats a good idea... The only problem is that if is not put in the correct place it might be hidden by other modals. Might need to tinker with it to make sure the Z prop is set